pimd/pim_addr.h

```diff
@@ -122,6 +122,14 @@ static inline uint32_t pim_sgaddr_hash(const pim_sgaddr a, uint32_t initval)
 	return jhash2((uint32_t *)&a, sizeof(a) / sizeof(uint32_t), initval);
 }
 
+static inline void pim_addr_from_ipaddr(pim_addr *ret, struct ipaddr ip)
+{
+	if (IS_IPADDR_V4(&ip))
```

Comment:
Hm. We should check here if the `struct ipaddr` matches what we expect, otherwise we could end up with weird invalid addresses.

Reply:
Actually since this is only used with `yang_dnode_get_ip` I don't think we need it at all if we have `yang_dnode_get_pimaddr` instead.

Reply:
Yeah, fixed it.